modules/dag-l1/src/main/scala/org/tessellation/dag/l1/infrastructure/block/rumor/handler.scala

```diff
@@ -0,0 +1,21 @@
+package org.tessellation.dag.l1.infrastructure.block.rumor
+
+import cats.effect.Async
+import cats.effect.std.Queue
+
+import org.tessellation.dag.domain.block.DAGBlock
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.gossip.ReceivedRumor
+import org.tessellation.sdk.infrastructure.gossip.RumorHandler
+import org.tessellation.security.signature.Signed
+
+object handler {
+
+  def blockRumorHandler[F[_]: Async: KryoSerializer](
+    peerBlockQueue: Queue[F, Signed[DAGBlock]]
+  ): RumorHandler[F] =
+    RumorHandler.fromReceivedRumorFn[F, Signed[DAGBlock]]() {
+      case ReceivedRumor(_, signedBlock) =>
+        peerBlockQueue.offer(signedBlock)
```

Comment:
I still suggest to make it through a storage/service instead directly on the queue :D

Reply:
If I make it through the storage I won't be able to stop the single stream with a semaphore and perform redownload e.g. That was my idea here.

Reply:
At least not as easily as I intended it to be :)

Reply:
why? you shouldn't stop a queue itself but a service that pulls data